### PR TITLE
RCLOUD-601: Adding job start failure metric.

### PR DESCRIPTION
**Pull request by jbrookspd**
Adding a jmx metric meter to measure job start failures.

### CODE REVIEW

**Review by sjrd218 — APPROVED** [review]

File: rundeckapp/grails-app/jobs/rundeck/quartzjobs/ExecutionJob.groovy [hunk 1]
```diff
@@ -173,6 +175,13 @@ class ExecutionJob implements InterruptableJob {
         initMap.jobSchedulerService.afterExecution(initMap.execution.asReference(), context.mergedJobDataMap, initMap.authContext)
     }
 
+    private void markStartExecutionFailure(JobExecutionContext context) {
+        MetricRegistry metricRegistry=context.jobDetail.jobDataMap.get('metricRegistry')
+        if(metricRegistry) {
+            metricRegistry.meter(MetricRegistry.name(this.class.name, 'executionJobStartFailedMeter')).mark()
```
Comment (gschueler) [inline rundeckapp/grails-app/jobs/rundeck/quartzjobs/ExecutionJob.groovy:181]: hmm, using `this.class.name` will now produce two different "metrics", do you want to consolidate them?

Comment (jbrookspd) [reply rundeckapp/grails-app/jobs/rundeck/quartzjobs/ExecutionJob.groovy:181]: Yea, that would make things easier. Is there an existing pattern for that or should I just pick one class name and use it?

Comment (gschueler) [reply rundeckapp/grails-app/jobs/rundeck/quartzjobs/ExecutionJob.groovy:181]: perhaps just use ExecutionService class name as the basis

**Review by gschueler — APPROVED** [review]

**Comment by jbrookspd** [comment]
This adds a new JMX metric, executionJobStartFailedMeter, in order to track the number of jobs start failures. 

Combined with the existing metric, executionJobStartMeter, it allows you to track the number of jobs that failed to start in contrast with the total number of jobs that were started.

**Pull request merged**